src/librustc/ty/maps.rs

```diff
@@ -96,9 +96,9 @@ impl<'tcx> Value<'tcx> for Ty<'tcx> {
     }
 }
 
-pub struct CycleError<'a, 'tcx: 'a> {
+pub struct CycleError<'tcx> {
     span: Span,
-    cycle: RefMut<'a, [(Span, Query<'tcx>)]>,
+    cycle: Vec<(Span, Query<'tcx>)>,
```

Comment:
Ohhh, you were talking about `item_path_str`, now I know exactly what you're talking about.
Yeah just do the `stack.to_vec(); drop(stack)` thing when reporting.